File: algorithms/cpp/119.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> getRow(int rowIndex)
    {
        vector<int> row(rowIndex+1, 0);
        row[0] = 1;
        for ( int i = 1; i <= rowIndex; i++ )
            for ( int j = i; j > 0; j-- )
                row[j] += row[j-1];
        for ( int i = 0; i < rowIndex/2; i++ )
            row[rowIndex-i] = row[i];
        return row;
    }
};

int main()
{
    Solution solution;
    for ( int k = 0; k < 5; k++ )
    {
        vector<int> result = solution.getRow(k);
        for ( int i = 0; i < (int)result.size(); i++ )
            cout << result[i] << " ";
        cout << endl;
    }
    return 0;
}

File: algorithms/rust/119.rs
impl Solution {
    #[allow(dead_code)]
    pub fn get_row(row_index: i32) -> Vec<i32> {
        let mut last_row = vec![1];
        for _ in 0..row_index {
            let mut row = vec![1; last_row.len() + 1];
            for i in 1..last_row.len() {
                row[i] = last_row[i - 1] + last_row[i];
            }
            last_row = row;
        }
        last_row
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
